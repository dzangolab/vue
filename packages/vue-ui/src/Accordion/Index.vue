<template>
  <slot name="title"></slot>
  <ul
    v-if="$slots.default"
    :class="props.direction"
    :aria-orientation="props.direction"
    class="accordion"
  >
    <li
      v-for="(slot, index) in $slots.default()"
      :key="index"
      :class="{ active: index === props.defaultIndex }"
    >
      <button type="button" @click="handleClick(index)">
        <img v-if="slot.props?.icon" :src="slot.props.icon" />
        <span>{{ slot.props ? slot.props.title : "" }}</span>

        <img
          v-if="props.activeIcon"
          :src="index === active ? props.activeIcon : props.inactiveIcon"
        />
      </button>

      <div v-if="index === active">
        <component :is="slot" />
      </div>
    </li>
  </ul>
</template>

<script lang="ts">
export default {
  name: "Accordion",
};
</script>

<script lang="ts" setup>
import { ref } from "vue";

import type { PropType } from "vue";

const props = defineProps({
  activeIcon: {
    default: "",
    required: false,
    type: String,
  },
  canSelfCollapse: {
    default: false,
    required: false,
    type: Boolean,
  },
  defaultIndex: {
    default: 0,
    required: false,
    type: Number,
  },
  direction: {
    default: "vertical",
    required: false,
    type: String as PropType<"vertical" | "horizontal">,
  },
  inactiveIcon: {
    default: "",
    required: false,
    type: String,
  },
});

const active = ref(props.defaultIndex);

const handleClick = (index: number) => {
  if (!props.canSelfCollapse || active.value !== index) {
    active.value = index;
  } else {
    active.value = -1;
  }
};
</script>

<style scoped>
.accordion {
  display: flex;
  list-style-type: none;
  margin: 0rem;
  padding: 0rem;
}

.accordion > li {
  display: flex;
}

.accordion > li > button {
  align-items: center;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  gap: 0.5rem;
  padding: 1rem;
  width: 100%;
}

.accordion.horizontal > li > button {
  flex-direction: column;
}

.accordion.horizontal {
  align-items: stretch;
  flex-direction: row;
}

.accordion.horizontal > li {
  flex-direction: row;
}

.accordion.horizontal > li > button {
  align-items: center;
  flex-direction: column;
  justify-content: space-between;
  transform: rotate(180deg);
  width: auto;
}

.accordion.horizontal > li > button > span {
  text-orientation: sideways-right;
  writing-mode: vertical-rl;
}

.accordion.horizontal > li > button > img:first-child {
  transform: rotate(90deg);
}

.accordion.horizontal > li > button > img:last-child {
  transform: rotate(270deg);
}

.accordion.vertical,
.accordion.vertical > li {
  flex-direction: column;
  width: 100%;
}

.accordion.vertical > li > button > img:last-child {
  margin-left: auto;
}
</style>
