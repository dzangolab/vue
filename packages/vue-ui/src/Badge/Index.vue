<template>
  <span :class="badgeClassName">
    <span v-if="iconLeft || !!slots.iconLeft" class="icon-left">
      <slot name="iconLeft">
        <i :class="iconLeft" />
      </slot>
    </span>

    <div v-if="label" class="label">
      {{ label }}
    </div>

    <span v-if="iconRight || !!slots.iconRight" class="icon-right">
      <slot name="iconRight">
        <i :class="iconRight" />
      </slot>
    </span>
  </span>
</template>

<script lang="ts">
export default {
  name: "BadgeComponent",
};
</script>

<script setup lang="ts">
import { computed, useSlots } from "vue";

const props = defineProps({
  className: {
    default: "",
    type: String,
  },
  color: {
    default: "default",
    type: String,
  },
  iconLeft: {
    default: null,
    type: String,
  },
  iconRight: {
    default: null,
    type: String,
  },
  label: {
    default: null,
    type: [String, Boolean],
  },
  rounded: Boolean,
  severity: {
    default: "primary",
    type: String,
    validator: (value: string) =>
      [
        "primary",
        "secondary",
        "alternate",
        "success",
        "danger",
        "warning",
      ].includes(value),
  },
});

const slots = useSlots();

const badgeClassName = computed(() => {
  return ["badge", props.className, props.rounded && "rounded", props.severity]
    .filter(Boolean)
    .join(" ");
});
</script>
