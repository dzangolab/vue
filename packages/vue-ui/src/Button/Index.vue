<template>
  <component
    :is="to ? 'a' : 'button'"
    :aria-label="ariaLabel ?? label"
    :class="buttonClassName"
    :disabled="!to && isDisabled"
    :href="to && !isDisabled ? to : null"
    :role="buttonRole"
    @click="!to && handleClick()"
  >
    <span v-if="iconLeft || !!slots.iconLeft" class="icon-left">
      <slot name="iconLeft">
        <i :class="iconLeft" />
      </slot>
    </span>

    <div v-if="children || label" class="label">
      <slot name="buttonContent">
        <span>{{ label }}</span>
      </slot>
    </div>

    <span v-if="iconRight || !!slots.iconRight" class="icon-right">
      <slot name="iconRight">
        <i :class="iconRight" />
      </slot>
    </span>

    <LoadingIcon v-if="loading" class="loading-button" />
  </component>
</template>

<script lang="ts">
export default {
  name: "ButtonElement",
};
</script>

<script setup lang="ts">
import { computed, useSlots } from "vue";

import LoadingIcon from "../components/LoadingIcon.vue";

const emits = defineEmits(["click"]);

const props = defineProps({
  ariaLabel: {
    default: "button",
    type: String,
  },
  children: {
    default: false,
    type: Boolean,
  },
  className: {
    default: "",
    type: String,
  },
  disabled: {
    default: false,
    type: Boolean,
  },
  iconLeft: {
    default: null,
    type: [String, Boolean],
  },
  iconRight: {
    default: null,
    type: [String, Boolean],
  },
  label: {
    default: null,
    type: String,
  },
  loading: {
    default: false,
    type: Boolean,
  },
  rounded: {
    default: false,
    type: Boolean,
  },
  severity: {
    default: "primary",
    type: String,
    validator: (value: string) =>
      [
        "primary",
        "secondary",
        "alternate",
        "success",
        "danger",
        "warning",
      ].includes(value),
  },
  size: {
    default: "medium",
    type: String,
    validator: (value: string) => ["small", "medium", "large"].includes(value),
  },
  to: {
    default: null,
    type: String,
  },
  title: {
    default: null,
    type: String,
  },
  variant: {
    default: "filled",
    type: String,
    validator: (value: string) =>
      ["outlined", "filled", "textOnly"].includes(value),
  },
});

const slots = useSlots();

const buttonClassName = computed(() => {
  return [
    "button",
    props.className,
    props.severity,
    props.size,
    props.variant,
    !(props.label || props.children) && "icon-only",
    props.loading && "loading",
    props.rounded && "rounded",
    isDisabled.value && "disabled",
  ]
    .filter(Boolean)
    .join(" ");
});

const buttonRole = computed(() => (props.to ? "link" : "button"));

const isDisabled = computed(() => props.disabled || props.loading);

function handleClick() {
  emits("click");
}
</script>

<style lang="css">
.button .loading-button {
  font-size: 4px;
  position: absolute;
  z-index: 999;
}
</style>
