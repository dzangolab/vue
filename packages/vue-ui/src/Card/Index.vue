<template>
  <div class="card">
    <header v-if="$slots.header || title" class="card-header">
      <slot name="header">
        <h3 class="card-title">
          {{ title }}
        </h3>
      </slot>
    </header>

    <div class="card-body">
      <slot></slot>
    </div>

    <footer v-if="$slots.footer" class="card-footer">
      <slot name="footer"></slot>
    </footer>
  </div>
</template>

<script lang="ts">
export default {
  name: "Card",
};
</script>

<script setup lang="ts">
defineProps({
  title: {
    type: String,
    default: undefined,
  },
});
</script>
