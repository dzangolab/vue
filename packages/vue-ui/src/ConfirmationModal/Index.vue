<template>
  <div :class="modalClassName">
    <div class="modal-wrapper" @click="handleClose()">
      <div class="modal-container">
        <div class="modal-header">
          <slot v-if="!disableHeader" name="header">
            <span>Confirmation required</span>
          </slot>

          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="1em"
            height="1em"
            viewBox="0 0 16 16"
            @click="handleClose()"
          >
            <path
              fill="currentColor"
              d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
            />
          </svg>
        </div>

        <div v-if="!disableBody" class="modal-body">
          <slot name="body">
            <p>Are you sure you want to proceed?</p>
          </slot>
        </div>

        <div v-if="!disableFooter" class="modal-footer">
          <slot name="footer">
            <ButtonElement
              size="small"
              label="Cancel"
              severity="secondary"
              @click="handleClose()"
            />

            <ButtonElement
              size="small"
              label="Confirm"
              severity="primary"
              @click="handleConfirm()"
            />
          </slot>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
export default {
  name: "ConfirmationModal",
};
</script>

<script setup lang="ts">
import { computed } from "vue";

import ButtonElement from "../Button/Index.vue";

const emits = defineEmits(["on:confirm", "on:close"]);

const props = defineProps({
  border: {
    default: false,
    type: Boolean,
  },
  disableBody: {
    default: false,
    type: Boolean,
  },
  disableHeader: {
    default: false,
    type: Boolean,
  },
  disableFooter: {
    default: false,
    type: Boolean,
  },
});

const modalClassName = computed(() => {
  return ["modal", props.border && "bordered-modal"].filter(Boolean).join(" ");
});

function handleConfirm() {
  emits("on:confirm");
}

function handleClose() {
  emits("on:close");
}
</script>
