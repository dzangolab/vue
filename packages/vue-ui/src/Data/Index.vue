<template>
  <div :class="`data direction-${direction}`">
    <div class="data-label">{{ label }}</div>
    <div v-if="slots.separator" class="separator">
      <slot name="separator"></slot>
    </div>
    <div class="data-value">
      <slot name="value">
        {{ displayValue }}
      </slot>
    </div>
  </div>
</template>

<script lang="ts">
export default {
  name: "DataElement",
};
</script>

<script setup lang="ts">
import { computed, useSlots } from "vue";

import type { PropType } from "vue";

const props = defineProps({
  dataKey: {
    default: undefined,
    type: String,
  },
  direction: {
    default: "vertical",
    type: String,
    validator: (value: string) => ["horizontal", "vertical"].includes(value),
  },
  label: {
    type: [String, Number, Object] as PropType<string | number>,
    required: true,
  },
  value: {
    default: "",
    type: [Object, String, Number] as PropType<string | number | object>,
  },
});

const slots = useSlots();

const displayValue = computed(() => {
  if (
    props.dataKey &&
    typeof props.value === "object" &&
    props.value !== null &&
    props.dataKey in props.value
  ) {
    return props.value[props.dataKey as keyof typeof props.value];
  }
  return props.value;
});
</script>

<style lang="css">
@import "../assets/css/data.css";
</style>
