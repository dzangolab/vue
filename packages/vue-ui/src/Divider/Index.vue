<template>
  <div :data-aria-orientation="orientation" class="divider" role="separator" />
</template>

<script lang="ts">
export default {
  name: "DividerElement",
};
</script>

<script setup lang="ts">
defineProps({
  orientation: {
    default: "horizontal",
    type: String,
    validator: (value: string) => ["horizontal", "vertical"].includes(value),
  },
});
</script>

<style lang="css">
.divider[role="separator"] {
  --_border: var(--divider-border, 1px solid #d0d0d0);
}

.divider[role="separator"][data-aria-orientation="horizontal"] {
  border-top: var(--_border);
  height: 0;
  margin: 1rem 0;
  width: 100%;
}

.divider[role="separator"][data-aria-orientation="vertical"] {
  border-left: var(--_border);
  height: 100%;
  margin: 0 1rem;
  width: 0;
}
</style>
