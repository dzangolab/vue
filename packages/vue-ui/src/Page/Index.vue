<template>
  <div class="page">
    <h1 v-if="title">
      {{ title }}
      <small v-if="subTitle">{{ subTitle }}</small>
    </h1>

    <div class="toolbar">
      <slot name="toolbar"></slot>
    </div>

    <slot name="default"></slot>
  </div>
</template>

<script lang="ts">
export default {
  name: "Page",
};
</script>

<script setup lang="ts">
import type { PropType } from "vue";

defineProps({
  subTitle: {
    default: undefined,
    required: false,
    type: String as PropType<string>,
  },
  title: {
    default: undefined,
    required: false,
    type: String as PropType<string>,
  },
});
</script>

<style lang="css" scoped>
.page {
  display: flex;
  flex-direction: column;
}

.page .content {
  padding-top: 1rem;
  padding-bottom: 1rem;
  flex-grow: 2;
}
</style>
