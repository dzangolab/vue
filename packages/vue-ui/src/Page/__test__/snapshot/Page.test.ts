import { mount, RouterLinkStub } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import Index from "../../Index.vue";

describe("Page", () => {
  it("matches snapshot", () => {
    const wrapper = mount(Index, {
      global: {
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
      props: {
        title: "Home",
        subTitle: "Welcome to the home page",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
