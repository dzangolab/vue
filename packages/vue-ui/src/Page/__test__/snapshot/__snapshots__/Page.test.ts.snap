// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page > matches snapshot 1`] = `
<div
  class="page"
>
  <h1>
    Home 
    <small>
      Welcome to the home page
    </small>
  </h1>
  
  
  <main>
    
    
  </main>
</div>
`;
